spec.bs

```diff
@@ -916,9 +929,13 @@ as appropriate:
     [=upon rejection=] of |promise|, run the following steps:
     1. Let |globalScopes| be |this|'s [=Worklet/global scopes=].
     1. [=Assert=]: |globalScopes|' [=list/size=] equals 1.
-    1. Set |globalScopes|[0]'s {{SharedStorageWorkletGlobalScope/
-        privateAggregation}}'s [=PrivateAggregation/scoping details=] a new
-            [=scoping details=] with the items:
+    1. Let |privateAggregationObj| be |globalScopes|[0]'s
+        {{SharedStorageWorkletGlobalScope/privateAggregation}}.
+    1. If [=this=]'s [=relevant global object=]'s [=associated document=] is
+        [=allowed to use=] "<code>[=private-aggregation=]</code>", set
+        |privateAggregationObj|'s [=PrivateAggregation/allowed to use=] to true.
+    1. Set |privateAggregationObj|'s [=PrivateAggregation/scoping details=] a
```

Comment:
Shall we early return if it's not allowed to use?

Reply:
The current (implemented) behavior is actually to check the 'is initial execution' complete first, although we could consider swapping.

Reply:
Yeah, that's fair -- I've added an issue to track whether we want to swap the order. I'll follow up soon